### PR TITLE
sample: upgrade curated environment

### PR DESCRIPTION
**Pull request by elliotzh**
# Description

Upgrade deprecated curated environments for pipeline related samples.

Deprecated environment includes:
+ azureml:AzureML-sklearn-1.0-ubuntu20.04-py38-cpu:*
+ azureml:azureml-sklearn-1.0:*

New valid environment is 
**azureml://registries/azureml/environments/sklearn-1.0/labels/latest**

# Checklist

- [ ] I have read the contribution guidelines.
  - [General](https://github.com/Azure/azureml-examples/blob/main/CONTRIBUTING.md)
  - [SDK](https://github.com/Azure/azureml-examples/blob/main/sdk/CONTRIBUTING.md)
  - [CLI](https://github.com/Azure/azureml-examples/blob/main/cli/CONTRIBUTING.md)
- [ ] I have coordinated with the docs team (mldocs@microsoft.com) if this PR deletes files or changes any file names or file extensions.
- [ ] Pull request includes test coverage for the included changes.
- [ ] This notebook or file is added to the [CODEOWNERS](https://github.com/Azure/azureml-examples/blob/main/.github/CODEOWNERS) file, pointing to the author or the author's team.

### CODE REVIEW

**Review by vizhur — CHANGES_REQUESTED** [review]
there is more https://github.com/search?q=repo%3AAzure%2Fazureml-examples%20ubuntu18.04&type=code

**Review by fbsolo-ms1 — APPROVED** [review]

**Review by lgayhardt — APPROVED** [review]

**Comment by elliotzh** [comment]
Hi @vizhur
1. other samples are not owned by pipeline team
2. I have checked other usages, they want an image instead of a curated environment, so not sure if the fix you have suggested fit for them.

File: tutorials/e2e-ds-experience/e2e-ml-workflow.ipynb [hunk 1]
```diff
@@ -744,7 +744,7 @@
                 "code: .\n",
                 "environment:\n",
                 "  # for this step, we'll use an AzureML curate environment\n",
-                "  azureml:AzureML-sklearn-1.0-ubuntu20.04-py38-cpu:1\n",
+                "  azureml://registries/azureml/environments/sklearn-1.0/versions/27\n",
```
[the remark this answers is not in the document]
Comment (elliotzh) [reply tutorials/e2e-ds-experience/e2e-ml-workflow.ipynb:747]: good catch, will try

**Comment by vizhur** [comment]
> Hi @vizhur
> 
> 1. other samples are not owned by pipeline team
> 2. I have checked other usages, they want an image instead of a curated environment, so not sure if the fix you have suggested fit for them.

1. ok, please fix whatever owned by pipeline team
2. not sure what are you talking about

File: sdk/python/jobs/pipelines/2e_image_classification_keras_minist_convnet/image_classification_keras_minist_convnet.ipynb [hunk 2]
```diff
@@ -108,22 +108,52 @@
...
+    "In this sample, we used [`mnist`](https://learn.microsoft.com/en-us/azure/open-datasets/dataset-mnist?tabs=azure-storage) as the target task. You can also try [`fashion-mnist`](https://github.com/zalandoresearch/fashion-mnist?tab=readme-ov-file#get-the-data), which has the same structure as `mnist` but of a harder task.\n",
+    "\n",
+    "Note that, **you may meet failure in the below block if you're in a private network and have no access to the public blob**. If so, you can try downloading from [official site of `fashion-mnist`](https://github.com/zalandoresearch/fashion-mnist?tab=readme-ov-file#get-the-data) or another machine, and unzip the downloaded file to [mnist](./mnist/)."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {
-    "name": "define-input"
```
Comment (superyyrrzz) [inline sdk/python/jobs/pipelines/2e_image_classification_keras_minist_convnet/image_classification_keras_minist_convnet.ipynb:118]: @elliotzh azure-docs-pr is failing due to this update: https://github.com/MicrosoftDocs/azure-docs-pr/pulls Could you add it back?

**Pull request merged**